pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -285,6 +299,15 @@ func (c *EtcdProxyController) syncHandler(key string) error {
 	// attempt processing again later. This could have been caused by a
 	// temporary network failure, or any other transient reason.
 	if err != nil {
+		etcdstorage, updateErr := c.updateEtcdStorageStatus(etcdstorage, etcdstoragev1alpha1.EtcdStorageCondition{
+			Type:    etcdstoragev1alpha1.ReplicaSetFailure,
+			Status:  etcdstoragev1alpha1.ConditionTrue,
+			Reason:  "EtcdProxyReplicaSetError",
+			Message: "failed to create etcdproxy replicaset",
```

Comment:
Would it be okay to put `err` instead of the generic message? It could give some more information and context about the error. Not sure are there any downsides?

Reply:
usually it's about leaking information one does not want to be leaked. But as these EtcdStorage objects are cluster-wide anyway and only created by cluster-admins therefore, I would add the error here as well.